### DISCUSSION TITLE
ESMValTool Tech Lead Call April 2022

### DISCUSSION THREAD

**Discussion by valeriupredoi** [ESMValGroup/Community#20] [Meetings]
Hey guys,

I set up a pool for our next Tech Lead call that will be in April, please complete it by end of next week, at your convenience: https://doodle.com/meeting/participate/id/mepMlrXa

Cheers and a good weekend!
V

**Comment by valeriupredoi** [comment]
Hey guys @ESMValGroup/technical-lead-development-team 

Many thanks for filling the doodle poll, best time for all of us is:

**Thursday, April 7, 2-3pm London (GMT+1) so 3-4pm CET (European Union)**
Link to Zoom call: https://ncas.zoom.us/j/98657433883


Cheers to @zklaus for raising some tentative agenda points, here below, please feel free to add more pints, I mean points:

## Agenda

- Last call [minutes](https://github.com/ESMValGroup/Community/blob/main/Tech%20Lead/Minutes/20220211.md)
- https://github.com/ESMValGroup/ESMValCore/issues/1541
- Platform to take notes for this meeting, Klaus suggesting https://hackmd.io/@CS80aeLkSQOjqsG5kzUKqg/ByUP4nYQq
- Implementation of the new Contributor License Agreement (suggested by @hb326)
- Code quality check settings https://github.com/ESMValGroup/ESMValTool/pull/2527 (suggested by @bouweandela)

**Comment by valeriupredoi** [comment]
## Agenda

### Attendance

Birgit, Klaus, Bouwe, Manu, Saskia, Emma, Remi, V

### Topics

- https://github.com/ESMValGroup/ESMValCore/issues/154 
  - Manuel: all dependencies from Conda forge - outstanding issues: conda feedstock meta file has divergent packages list - lots from PyPi in `setup.py` - move all to env, careful with naming from PyPi vs CF - careful with non-Python deps (NCL, R stuff) 
  - so crack on with curr PR (just need to re-add deps in setup.py) and to the same for Tool
- CLA crappe: Birgit - the Consortium Agreement is almost done, CLA is being talked about - should have one single clickyty button on GH  
  - unless we have strong arguments against the CLA app suggested (CLA Auth)?
- Codacy discussion (Bouwe) https://github.com/ESMValGroup/ESMValTool/pull/2527 : keep it high for Core, don't lower it! Maybe lower it for Tool? 
  - Klaus: it gets very hairy to maintain stuff that's badly written in Tool - fix a standard, whichever we decide on, best to fix it - timescale determined by the versions of the tools we use;
  - Reset to max Errs to 0 - even big errors go through (Bouwe); Remi, Klaus agree - Manu: how do you decide which goes through? -> Technical reviewer of PR decides
  - **Vote** Tech team - voted to get it back to 0 yay
- Talk about big OBS files - open an issue for a better discussion; also, how to grab continually updated OBS data (and CMORize it and update our DB) - Remi - enough tech discussion )especially lots are interested in the DB - MO, IPSL etc - Birgit: user point of view data stopping at 2014 is good fro CMIP6 - Manu: lots of changes to cmorizer scripts stemmed from hardcoded, custom vars etc (eg version) - Bouwe: two diff probs - distribution **and** data cmorization - for distribution Emma is interested to get smth like an online DB a la obs4MIPS 0 Klaus: get an rsync pull from DKRZ with account credentials, sort of a stop gap solution -> we will talk more (maybe at the Workshop in June)
- Have us a Tech meeting at the Workshop in June?

### TODO's/Outstanding

- OBS database/transfer and CMORization
- large OBS data files
- fix the deps from conda-forge only via PR's (Core has one, needs fixing by V; Tool needs one too)

**Reply by bouweandela** [reply]
Thanks for organizing the meeting @valeriupredoi!

**Comment by valeriupredoi** [comment]
always a pleasure see yer faces, bud :grin:

**Comment by remi-kazeroni** [comment]
I have just checked the Codacy settings and I think someone already set the number of max issues to 0 👍 

Shall we open a Discussion for the Tool about that? That would help to inform the community about this decision. And that could be a place where we would record decisions made for _some types_ of controversial Codacy issues as we encounter them while reviewing PRs, not necessarily all types. For example: "too many local variables" -> "usually accept as is", "critical issue" -> "usually try to fix", ... I can start that if you want 👍

**Reply by bouweandela** [reply]
Great idea @remi-kazeroni! It would be great if you could start the discussion too. I usually take into account the programming skill level of the person making the pull request too when deciding what is acceptable.

**Reply by bouweandela** [reply]
And how much they (dis)like fixing technical stuff ;-)
